librz/core/cconfig.c

```diff
@@ -1737,6 +1737,180 @@ RZ_API bool rz_core_esil_cmd(RzAnalysisEsil *esil, const char *cmd, ut64 a1, ut6
+		}
+		free(s);
+		return true;
+	}
+	eval_string(core->config, s);
+	free(s);
+	return true;
+}
+
+static void config_print_node(RzConfig *cfg, RzConfigNode *node, RzCmdStateOutput *state) {
+	rz_return_if_fail(cfg);
+	char *option;
+	bool isFirst;
+	RzOutputMode mode = state->mode;
+	PJ *pj = state->d.pj;
+	RzListIter *iter;
+	char *es = NULL;
+
+	if (mode == RZ_OUTPUT_MODE_JSON) {
+		if (rz_str_isnumber(node->value)) {
+			pj_kn(pj, node->name, rz_num_math(NULL, node->value));
+			return;
+		} else if (rz_str_is_bool(node->value)) {
+			pj_kb(pj, node->name, (node->value));
+			return;
+		}
+		pj_ks(pj, node->name, node->value);
+	} else if (mode == RZ_OUTPUT_MODE_LONG_JSON) {
+		pj_o(pj);
+		pj_ks(pj, "name", node->name);
+		if (rz_str_isnumber(node->value)) {
+			pj_kn(pj, "value", rz_num_math(NULL, node->value));
+		} else if (rz_str_is_bool(node->value)) {
+			pj_kb(pj, "value", (node->value));
+		} else {
+			pj_ks(pj, "value", node->value);
+		}
+		pj_ks(pj, "type", rz_config_node_type(node));
+		es = rz_str_escape(node->desc);
+		if (es) {
+			pj_ks(pj, "desc", es);
+			free(es);
+		}
+		pj_kb(pj, "ro", rz_config_node_is_ro(node));
+		if (!rz_list_empty(node->options)) {
+			pj_ka(pj, "options");
+			rz_list_foreach (node->options, iter, option) {
+				pj_s(pj, option);
+			}
+			pj_end(pj);
+		}
+		pj_end(pj);
+	} else if (mode == RZ_OUTPUT_MODE_LONG) {
+		rz_cons_printf("%s = %s %s; %s",
+			node->name, node->value,
+			rz_config_node_is_ro(node) ? "(ro)" : "",
+			node->desc);
+		if (!rz_list_empty(node->options)) {
+			isFirst = true;
+			rz_cons_printf(" [");
+			rz_list_foreach (node->options, iter, option) {
+				if (isFirst) {
+					isFirst = false;
+				} else {
+					rz_cons_printf(", ");
+				}
+				rz_cons_printf("%s", option);
+			}
+			rz_cons_printf("]");
+		}
+		rz_cons_println("");
+	} else if (mode == RZ_OUTPUT_MODE_QUIET) {
+		rz_cons_printf("%s=%s\n", node->name, node->value);
+	} else if (mode == RZ_OUTPUT_MODE_RIZIN) {
+		rz_cons_printf("e %s=%s\n", node->name, node->value);
+	} else if (mode == RZ_OUTPUT_MODE_STANDARD) {
+		rz_cons_printf("%20s: %s\n", node->name,
+			node->desc ? node->desc : "");
+	}
+}
+
+RZ_API void rz_core_config_print_all(RzConfig *cfg, const char *str, RzCmdStateOutput *state) {
+	rz_return_if_fail(cfg);
+	RzConfigNode *node;
+	RzListIter *iter;
+	int len = 0;
+	PJ *pj = state->d.pj;
+	RzOutputMode mode = state->mode;
+
+	if (mode == RZ_OUTPUT_MODE_LONG_JSON) {
+		pj_a(pj);
+	} else if (mode == RZ_OUTPUT_MODE_JSON) {
+		pj_o(pj);
+	}
+
+	if (RZ_STR_ISNOTEMPTY(str)) {
+		str = rz_str_trim_head_ro(str);
+		len = strlen(str);
+		if (len > 0 && str[0] == ' ') {
+			str++;
+			len--;
+		}
+	}
+
+	rz_list_foreach (cfg->nodes, iter, node) {
+		if (!str || (!strncmp(str, node->name, len))) {
+			config_print_node(cfg, node, state);
+		}
+	}
+
+	if (mode == RZ_OUTPUT_MODE_LONG_JSON || mode == RZ_OUTPUT_MODE_JSON) {
+		pj_end(pj);
+		rz_cons_println(pj_string(pj));
```

Comment:
this is not going to be needed after i push a fix for RzCmd.

Reply:
@officialcjunior can be removed after the rebase on top of the latest "dev"